### ISSUE TITLE
ConfigmDialog: disable close on Esc 

### ISSUE THREAD

**Issue by gianluca-valentini** [vaadin/vaadin-confirm-dialog-flow#94]
I'm using Vaadin 13.0.13.

Is there a way to use ConfirmDialog and disable the close on esc button?

I don't want that user can close the Dialog using esc o clicking out of it. Is this possible?
Gianluca

**Comment by web-padawan** [comment]
You should be able to do it by setting a web component property:

```
getElement().setProperty("noCloseOnEsc", true);
```

For some reason this API is not publicly exposed on ConfirmDialog. Should be probably the same as with Dialog, which has `isCloseOnEsc` and `setCloseOnEsc` for it:

https://github.com/vaadin/vaadin-dialog-flow/blob/77acb3535333595f6836e88cc8540619cb6d2ae8/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java#L234-L257

**Comment by gianluca-valentini** [comment]
I I tried it but it does not work

![image](https://user-images.githubusercontent.com/3224592/63751828-66712080-c8b0-11e9-9977-a2363c4af9a0.png)

Clicking outside, the dialog does not close. 
Using ESC button, the dialog closes
Maybe there is something else to set?

**Comment by DiegoCardoso** [comment]
Hi @gianluca-valentini,

Confirm Dialog listens to the escape key events coming from the overlay, but it used to ignore when the Dialog prevents default, if `noCloseOnEsc` is set to `true`. 

That was fixed with a new API on the Confirm Dialog web component, adding the same `noCloseOnEsc` property to it and making sure to check whether default was prevented or not. 

Unfortunately, this change is still on [`1.2.0-alpha1`](https://github.com/vaadin/vaadin-confirm-dialog/releases/tag/v1.2.0-alpha1) version of the web component and it didn't make to the flow component yet.

**Comment by gianluca-valentini** [comment]
Ok. When do you think that it will be possible to have this new feature?

**Comment by gianluca-valentini** [comment]
HI, any news about this feature?
Currently I'm using V14.1.16 and this feature is not present yet

**Comment by gianluca-valentini** [comment]
Any news about this feature?

**Comment by DiegoCardoso** [comment]
Yes. The version [1.2.0](https://github.com/vaadin/vaadin-confirm-dialog/releases/tag/v1.2.0) for the WC containing this feature is released. That version will land on LTS for 14.3. However the API for the Flow component hasn't yet been implemented, but this issue is in our backlog and we intend to make it to be part of V14.3.

**Comment by gianluca-valentini** [comment]
Thanks a lot
